src/v/compat/cluster_compat.h

```diff
@@ -381,6 +387,12 @@ struct compat_check<cluster::topic_properties> {
         json_read(record_value_subject_name_strategy_compat);
         json_read(initial_retention_local_target_bytes);
         json_read(initial_retention_local_target_ms);
+        std::optional<ss::sstring> vcluster;
+        json::read_member(rd, "mpx_virtual_cluster_id", vcluster);
+        if (vcluster) {
+            obj.mpx_virtual_cluster_id = model::vcluster_id(
+              xid::from_string(*vcluster));
+        }
```

Comment:
Suggestion:  You can add:
```c++
inline void read_value(json::Value const& rd, model::vcluster_id& v) {
    ss::sstring xid_str;
    read_value(rd, xid_str);
    v = model::vcluster_id(xid::from_string(xid_str));
}
```

to `compat/json.h` and then you can maintain the same `json_read(mpx_virtual_cluster_id)` pattern